quint/src/effects/base.ts

```diff
@@ -43,6 +43,19 @@ export type Effect =
   | ArrowEffect
   | QuantifiedEffect
 
+/*
+ * An effect scheme, listing universally quantified variables (for names
+ * refering to Variables) and effect variables (for names refering to Effects)
+ */
+export type EffectScheme = {
+  /* The effect */
+  effect: Effect,
+  /* Universally quantified names refering to Effects */
+  effectVariables: Set<string>,
+  /* Universally quantified names refering to Variables */
+  variables: Set<string>,
+}
```

Comment:
Yeah, these names are very confusing. I need to find some other word for the `Variables` interface (maybe `Region`, `Registers`, `References`, idk), and also stop using `quantified` for other stuff not on `EffectScheme`. I didn't want to do it along with these changes to avoid polluting the diff or spending 1 day rebasing all of this.

I'll add an example for now. LMK if you have any name ideas.